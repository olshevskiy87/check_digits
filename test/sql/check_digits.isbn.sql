CREATE EXTENSION IF NOT EXISTS check_digits;

-- should be true
select check_digits_isbn('1451627289');
select check_digits_isbn('978-1-934356-98-2');

-- should be false
select check_digits_isbn('5-93286-093-8');
select check_digits_isbn('5-19-036976-X');

-- should fail with error
-- by length
select check_digits_isbn('978-5-271-1611-8');
select check_digits_isbn('978-0-752860436-5');
