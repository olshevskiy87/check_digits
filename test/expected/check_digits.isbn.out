CREATE EXTENSION IF NOT EXISTS check_digits;
NOTICE:  extension "check_digits" already exists, skipping
-- should be true
select check_digits_isbn('1451627289');
 check_digits_isbn 
-------------------
 t
(1 row)

select check_digits_isbn('978-1-934356-98-2');
 check_digits_isbn 
-------------------
 t
(1 row)

-- should be false
select check_digits_isbn('5-93286-093-8');
 check_digits_isbn 
-------------------
 f
(1 row)

select check_digits_isbn('5-19-036976-X');
 check_digits_isbn 
-------------------
 f
(1 row)

-- should fail with error
-- by length
select check_digits_isbn('978-5-271-1611-8');
ERROR:  ISBN must contain 10 or 13 digits
select check_digits_isbn('978-0-752860436-5');
ERROR:  ISBN must contain 10 or 13 digits
